components/quick_input.jsx

```diff
@@ -73,18 +88,59 @@ export default class QuickInput extends React.PureComponent {
         this.input = input;
     }
 
+    onClear = () => {
+        if (this.props.onClear) {
+            this.props.onClear();
+        }
+
+        this.value = '';
+    }
+
     render() {
-        const {value, inputComponent, ...props} = this.props;
+        const clearableTooltip = (
+            <Tooltip id={'InputClearTooltip'}>
+                <FormattedMessage
+                    id={'input.clear'}
+                    defaultMessage='Clear input'
+                />
+            </Tooltip>
+        );
+
+        const {value, inputComponent, clearable, ...props} = this.props;
 
         Reflect.deleteProperty(props, 'delayInputUpdate');
+        Reflect.deleteProperty(props, 'onClear');
 
-        return React.createElement(
+        const inputElement = React.createElement(
             inputComponent || 'input',
             {
                 ...props,
                 ref: this.setInput,
                 defaultValue: value, // Only set the defaultValue since the real one will be updated using componentDidUpdate
             }
         );
+
+        return (<div>
+            {inputElement}
+            {clearable && this.input && this.value && this.value.trim() !== '' &&
```

Comment:
Let's remove this `trim` based on UX response, the `x` should appear even if there is only whitespaces.

Reply:
I believe this is the only outstanding concern, should be good to go after this @larkox! :)